pkg/kube/client.go

```diff
@@ -550,14 +550,40 @@ func perform(infos Result, fn ResourceActorFunc) error {
 		return ErrNoObjectsVisited
 	}
 
-	for _, info := range infos {
-		if err := fn(info); err != nil {
+	errs := make(chan error)
+	go batchPerform(infos, fn, errs)
+
+	for range infos {
+		err := <-errs
+		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func batchPerform(infos Result, fn ResourceActorFunc, errs chan<- error) {
+	finished := make(chan bool, 10000)
+	kind := infos[0].Object.GetObjectKind().GroupVersionKind().Kind
```

Comment:
Can we add a length check prior to this call? If `infos` is an empty list, this will result in a `panic`. We should probably `return` early in the case where there's no `infos` to process.

Reply:
Good call. Although `func perform` checks before calling batchPerform, this will be appropriate if it is ever reused.